### PR TITLE
chore: rename km_kbp_ to km_core_

### LINKED ISSUE #9722: chore(core): rename `km_kbp_` prefix to `km_core_` prefix in keyman_core_api.h
Depends on #9721.

For similar reasons as in #9721, moving to `km_core_` helps to clarify the interface boundaries.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9722.

@keymanapp-test-bot skip

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -926,7 +926,7 @@
 * chore: git tag with release@semver (#8035)
 * fix(linux): Properly set context after changing IP (#8026)
 * chore(linux): use faster zero-length string check (#8037)
-* chore(linux): log failures to `km_kbp_context_clear(context)` (#8036)
+* chore(linux): log failures to `km_core_context_clear(context)` (#8036)
```
Comment (darcywong00) [inline HISTORY.md:929]: Do we want to leave HISTORY.md intact? 
This PR would then add an entry for the rename.

Comment (mcdurdin) [reply HISTORY.md:929]: Yes my bad I backed out one change to HISTORY.md but missed the other one!

**Review by darcywong00 — APPROVED** [review]
lgtm

File: core/src/kmx/kmx_environment.cpp [hunk 2]
```diff
@@ -20,31 +20,31 @@ namespace {
...
-  Set(KM_KBP_KMX_ENV_PLATFORM, DEFAULT_PLATFORM);
-  Set(KM_KBP_KMX_ENV_BASELAYOUT, DEFAULT_BASELAYOUT);
-  Set(KM_KBP_KMX_ENV_BASELAYOUTALT, DEFAULT_BASELAYOUTALT);
-  Set(KM_KBP_KMX_ENV_SIMULATEALTGR, DEFAULT_SIMULATEALTGR);
-  Set(KM_KBP_KMX_ENV_BASELAYOUTGIVESCTRLRALTFORRALT, DEFAULT_BASELAYOUTGIVESCTRLRALTFORRALT);
+  Set(KM_CORE_KMX_ENV_PLATFORM, DEFAULT_PLATFORM);
+  Set(KM_CORE_KMX_ENV_BASELAYOUT, DEFAULT_BASELAYOUT);
+  Set(KM_CORE_KMX_ENV_BASELAYOUTALT, DEFAULT_BASELAYOUTALT);
+  Set(KM_CORE_KMX_ENV_SIMULATEALTGR, DEFAULT_SIMULATEALTGR);
+  Set(KM_CORE_KMX_ENV_BASELAYOUTGIVESCTRLRALTFORRALT, DEFAULT_BASELAYOUTGIVESCTRLRALTFORRALT);
```
Comment (rc-swag) [inline core/src/kmx/kmx_environment.cpp:27]: We could trim these down too (maybe seperate PR) `KM_CORE_KMX_ENV_PLATFORM` =>  `KM_CORE_KMX_PLATFORM`
or even drop the `KMX_` also

File: HISTORY.md (hunk 1 above)
Comment (rc-swag) [reply HISTORY.md]: I like this suggestion.

File: core/src/kmx/kmx_environment.cpp (hunk 2 above)
Comment (rc-swag) [reply core/src/kmx/kmx_environment.cpp:27]: otherwise lgtm

Comment (mcdurdin) [reply core/src/kmx/kmx_environment.cpp:27]: We could ... but the longer name clarifies that these are really only used by the KMX environment at present.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.187-alpha](https://keyman.com/downloads/releases/alpha/17.0.187)

**Pull request merged**